pkgs/development/interpreters/python/cpython/default.nix

```diff
@@ -310,12 +310,28 @@ in with passthru; stdenv.mkDerivation (finalAttrs: {
     mingw-patch = fetchgit {
       name = "mingw-python-patches";
       url = "https://src.fedoraproject.org/rpms/mingw-python3.git";
-      rev = "45c45833ab9e5480ad0ae00778a05ebf35812ed4"; # for python 3.11.5 at the time of writing.
-      sha256 = "sha256-KIyNvO6MlYTrmSy9V/DbzXm5OsIuyT/BEpuo7Umm9DI=";
+      rev = "3edecdbfb4bbf1276d09cd5e80e9fb3dd88c9511"; # for python 3.11.9 at the time of writing.
+      hash = "sha256-kpXoIHlz53+0FAm/fK99ZBdNUg0u13erOr1XP2FSkQY=";
     };
-  in [
-    "${mingw-patch}/*.patch"
```

Comment:
It worked when I merged https://github.com/NixOS/nixpkgs/pull/227818 and the last time I tried to compile python3 mingw (with patch conflicts) about a month ago.
Did the nix update from 2.18 to 2.24 break it?

Reply:
I'm trying to run a `git bisect` between master and that commit, but it's not easy to say. A few of the builds are failing because of `mpdecimal` having been a build req at one point, and it not being available on mingw. Let's see what the result of the bisect is, though, to know if it helps or if that false negative is going to interrupt it.

Reply:
It must've broke within 30 days

Reply:
Doing a `git bisect` between the current head of master and the one you linked above (ignoring the failures from `mpdecimal` being flagged non-Windows) brings me to this commit:
```
commit 6c04557835ebfd0fa90dd5037b7422a9d4300a81 (HEAD)
Merge: fba309020fb2 fe944df56f83
Author: Vladimír Čunát <v@cunat.cz>
Date:   Sun Oct 13 09:45:55 2024 +0200

    staging-next 2024-09-21 (#343421)
```

That doesn't seem like such a very helpful commit. I did run the bisect with `--first-parent`.

Reply:
> It must've broke within 30 days

[This is the commit](https://github.com/NixOS/nixpkgs/commit/bc0395ee6f9964065665adf3d365a2f93c3b4ca5) where my bisect of the staging branch shows the breakage appearing.